jung-visualization/src/main/java/edu/uci/ics/jung/visualization/renderers/CachingVertexRenderer.java

```diff
@@ -33,39 +34,36 @@ public CachingVertexRenderer(BasicVisualizationServer<V, E> vv) {
   }
 
   /** Paint <code>v</code>'s icon on <code>g</code> at <code>(x,y)</code>. */
-  protected void paintIconForVertex(V v) {
+  protected void paintIconForVertex(
+      RenderContext<V, E> renderContext, LayoutMediator<V, E> layoutMediator, V v) {
     GraphicsDecorator g = renderContext.getGraphicsContext();
-    boolean vertexHit = true;
+
     int[] coords = new int[2];
     Shape shape = vertexShapeMap.get(v);
     if (shape == null || dirtyVertices.contains(v)) {
-      shape = prepareFinalVertexShape(v, coords);
+      shape = prepareFinalVertexShape(renderContext, layoutMediator, v, coords);
       vertexShapeMap.put(v, shape);
       dirtyVertices.remove(v);
     }
-    vertexHit = vertexHit(shape);
-    if (vertexHit) {
-      if (renderContext.getVertexIconTransformer() != null) {
-        Icon icon = renderContext.getVertexIconTransformer().apply(v);
-        if (icon != null) {
+    if (renderContext.getVertexIconTransformer() != null) {

```

Comment:
The SpatialGrid makes it unnecessary to check for an intersection of the shape with the view here.